script/versions.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+"""
+Query the github API for the git tags of a project, and return a list of
+version tags for recent releases, or the default release.
+
+The default release is the most recent non-RC version.
+
+Recent is a list of unqiue major.minor versions, where each is the most
+recent version in the series.
+
+For example, if the list of versions is:
+
+    1.8.0-rc2
+    1.8.0-rc1
+    1.7.1
+    1.7.0
+    1.7.0-rc1
+    1.6.2
+    1.6.1
+
+`default` would return `1.7.1` and
+`recent -n 3` would return `1.8.0-rc2 1.7.1 1.6.2`
+"""
+from __future__ import print_function
+
+import argparse
+import itertools
+import operator
+from collections import namedtuple
+
+import requests
+
+
+GITHUB_API = 'https://api.github.com/repos'
+
+
+class Version(namedtuple('_Version', 'major minor patch rc')):
+
+    @classmethod
+    def parse(cls, version):
+        version = version.lstrip('v')
+        version, _, rc = version.partition('-')
+        major, minor, patch = version.split('.', 3)
+        return cls(int(major), int(minor), int(patch), rc)
+
+    @property
+    def major_minor(self):
+        return self.major, self.minor
+
+    @property
+    def order(self):
+        return self[:3] + (None, self.rc) if self.rc else ('', )
+
+    def __str__(self):
+        rc = '-{}'.format(self.rc) if self.rc else ''
+        return '.'.join(map(str, self[:3])) + rc
+
+
+def group_versions(versions):
+    return list(
+        list(releases)
```

Comment:
lists aren't unique!

No, I'm using lists so that I can index into the sequence. `itertools.groupby()` returns a generator which can't be indexed.

I really dislike adding unnecessary code. This is pretty succinct and uses clear variables names.  One of the great features of python is that you don't have to write overly verbose code to express your ideas. Every additional line of code is an opportunity for a bug to exist.  Fewer lines of code is fewer bugs!  I understand the need for clear variable names, and whitespace to separate ideas, but I think this function is better expressed in a functional way (as it is written) than expanding it out into multiple lines.

This function reads pretty well to me:

"List the releases for all versions grouped by the major_minor attribute".